packages/connect-popup/e2e/support/helpers.ts

```diff
@@ -131,9 +131,23 @@ export const downloadLogs = async (logPage: Page, downloadLogPath: string) => {
     return download;
 };
 
-export const setTrustedHost = async (explorerPage: Page, explorerUrl: string) => {
+export const setConnectSettings = async (
+    explorerPage: Page,
+    explorerUrl: string,
+    { trustedHost = false, connectSrc }: { trustedHost?: boolean; connectSrc?: string },
+) => {
     await explorerPage.goto(`${explorerUrl}#/settings`);
-    await waitAndClick(explorerPage, ['@checkbox/trustedHost']);
+    // TODO: We currently do not have a way to wait for handshake with service worker in webextension so
+    // TODO: now the best thing we have it waiting but we should change it once we have better way.
```

Comment:
really?

Reply:
Yes I know it is terrible, but this is just kind of intermediary step, I am thinking about some changes in the setting page that will allow the test to see the handshake is done so we do not have to do this terrible workaround.

Reply:
ok, so maybe would you create an issue to remove this? I am ok with merging it but we should fix this asap, this long wait is a smell

Reply:
also tagging @martykan maybe he gets some idea

Reply:
I am implementing it currently, commit coming soon.

Reply:
I think it makes sense to fix it before merging so I added 3 commits:

* https://github.com/trezor/trezor-suite/pull/10988/commits/dec5ccf5475dc5849894a4f9215e857314c2c59f Add isHandshakeConfirmed to settings page so we can use it in tests when webextension
* https://github.com/trezor/trezor-suite/pull/10988/commits/d0892424e106d048ced0f137ea45588538379666 - emit messages from proxy so it can be listened
* https://github.com/trezor/trezor-suite/pull/10988/commits/92e2015920bbb0ec0f115b53ff11567a6c3b169c - make test wait for selector so it continues when handshake is done.